### PR TITLE
Issue #189: replace JAX G naming with neg_efe

### PR DESCRIPTION
**Pull request by conorheins**
## Summary
- Replace public JAX naming from `G` to `neg_efe` while preserving the same numerics (`neg_efe = -EFE`).
- Remove `info["G"]` from `rollout()` extras and emit only `info["neg_efe"]`.
- Keep SPM homage in docs via explicit mapping: `SPM G = neg_efe = -EFE`.

## What changed
- `Agent.infer_policies(...)` docs/examples/locals now use `neg_efe`.
- `pymdp.control` internals renamed from `compute_G_*` / `neg_G` to `compute_neg_efe_*` / `neg_efe`.
- Sophisticated inference planning (`planning/si.py`) renamed internal `Tree.G` field and related recursive terms to `neg_efe`.
- Planning visualizer updated to read/display `neg_efe` keys.
- Docs updated in quickstart/migration/API/release notes; release notes include breaking key rename `G -> neg_efe`.
- Tests updated to assert rollout key replacement and renamed variables; added explicit sign-convention test for policy softmax equivalence.

## Validation
- `.venv311/bin/python -m pytest -q test/test_agent_jax.py test/test_categorical_observations.py test/test_rollout_function.py test/test_sophisticated_inference_jax.py test/test_param_info_gain_jax.py test/test_hmm_associative_scan.py`
- Result: 85 passed

### CODE REVIEW

File: pymdp/planning/si.py [hunk 1]
```diff
@@ -1107,13 +1121,13 @@ def continue_expansion(tree: Tree, agent: Any, h: int) -> jnp.ndarray:
...
-        #                 efestop=tree.G[tree.root_idx(), 0] >= efe_stop_threshold, qpi=jnp.all(q_pi == 0))
+        #                 estop=entropy <= entropy_stop_threshold, g=tree.neg_efe[tree.root_idx(), 0], efet=efe_stop_threshold,
+        #                 efestop=tree.neg_efe[tree.root_idx(), 0] >= efe_stop_threshold, qpi=jnp.all(q_pi == 0))
 
         return (
             (h < horizon)
             & (jnp.all(q_pi == 0) | (entropy > entropy_stop_threshold))
-            & (tree.G[root_idx(tree), 0] < efe_stop_threshold)
+            & (tree.neg_efe[root_idx(tree), 0] < efe_stop_threshold)
         )
```
[the remark this answers is not in the document]
Comment (conorheins) [reply pymdp/planning/si.py:1136]: Good catch. Fixed by replacing with `neg_efe_stop_threshold`

File: pymdp/control.py [hunk 2]
```diff
@@ -507,6 +524,12 @@ def compute_G_policy(
...
     """Compute policy-wise negative expected free energy for one policy.
 
+    Notes
+    -----
+    This function computes `neg_efe = -EFE` for a single policy. The function
+    name follows SPM-style notation where this policy score is commonly called
+    `G`.
+
     Parameters
     ----------
```
[the remark this answers is not in the document]
Comment (conorheins) [reply pymdp/control.py]: Agreed. Reworded docstrings in compute_neg_efe_policy and compute_neg_efe_policy_inductive to describe SPM mapping without implying Python names follow SPM naming.

File: pymdp/control.py [hunk 3]
```diff
@@ -585,6 +608,12 @@ def compute_G_policy_inductive(
     """
     Compute policy-wise negative expected free energy with inductive planning.
 
+    Notes
+    -----
+    This function computes `neg_efe = -EFE` for a single policy with optional
+    inductive-value terms. The function name follows SPM-style notation where
+    this score is commonly called `G`.
```
[the remark this answers is not in the document]
Comment (conorheins) [reply pymdp/control.py]: Agreed. Reworded docstrings in compute_neg_efe_policy and compute_neg_efe_policy_inductive to describe SPM mapping without implying Python names follow SPM naming.

File: pymdp/control.py [hunk 4]
```diff
@@ -676,6 +705,11 @@ def update_posterior_policies_inductive(
     """
     Compute policy posterior and expected free energy with optional inductive terms.
 
+    Notes
+    -----
+    The returned policy score is `neg_efe = -EFE`. In SPM-style notation this
+    same quantity is often denoted by `G`.
+
```
[the remark this answers is not in the document]
Comment (conorheins) [reply pymdp/control.py:712]: updated update_posterior_policies_inductive summary to explicitly say negative expected free energy.

File: pymdp/planning/si.py [hunk 5]
```diff
@@ -623,7 +626,7 @@ def optimized_tree_search(
 
     - planning horizon is reached
     - root policy entropy drops below `entropy_stop_threshold`
-    - root expected free energy drops below `efe_stop_threshold`
+    - root negative expected free energy drops below `neg_efe_stop_threshold`
```
[the remark this answers is not in the document]
Comment (conorheins) [reply pymdp/planning/si.py:632]: Updated SI docs to match current inequality semantics: expansion stops when root neg_efe >= neg_efe_stop_threshold.

File: docs-mkdocs/development/release-notes.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# Release Notes
+
+## 1.0.0 (in progress)
+
+- Breaking (JAX API): `rollout()` extras now use `info["neg_efe"]` instead of
+  `info["G"]`.
+- JAX examples and docs now use `q_pi, neg_efe = agent.infer_policies(qs)` for
+  policy-score outputs.
+- SPM notation mapping is now explicit in docs: SPM's `G` corresponds to
+  `neg_efe = -EFE`.
```
[the remark this answers is not in the document]
Comment (conorheins) [reply docs-mkdocs/development/release-notes.md:13]: Added release-note entry for the breaking SI keyword rename from efe_stop_threshold to neg_efe_stop_threshold.
